packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -0,0 +1,313 @@
+        const previousItemIndex: number = currentItemIndex > 0 ? currentItemIndex - 1 : items.length - 1;
+        const previousItemId: string = items[previousItemIndex].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId: previousItemId
+            });
+
+            (Array.from(this.tabListRef.current.children)[previousItemIndex] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(previousItemId);
+        }
+    }
+
+    /**
+     * Activates the next tab item
+     */
+    private activateNext(): void {
+        const items: JSX.Element[] = this.getChildBySlot(this.props.children, TabSlot.tabItem);
+        const currentItemIndex: number = items.findIndex(this.getCurrentIndexById);
+        const nextItemIndex: number = currentItemIndex < items.length - 1 ? currentItemIndex + 1 : 0;
+        const nextItemId: string = items[nextItemIndex].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId: nextItemId
+            });
+
+            (Array.from(this.tabListRef.current.children)[nextItemIndex] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(nextItemId);
+        }
+    }
+
+    /**
+     * Activates the first tab item
+     */
+    private activateFirst(): void {
+        const items: JSX.Element[] = this.getChildBySlot(this.props.children, TabSlot.tabItem);
+        const activeId: string = items[0].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId
+            });
+
+            (Array.from(this.tabListRef.current.children)[0] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(activeId);
+        }
+    }
+
+    /**
+     * Activates the last tab item
+     */
+    private activateLast(): void {
+        const items: JSX.Element[] = this.getChildBySlot(this.props.children, TabSlot.tabItem);
+        const lastItemIndex: number = items.length - 1;
+        const activeId: string = items[lastItemIndex].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId
+            });
+
+            (Array.from(this.tabListRef.current.children)[lastItemIndex] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(activeId);
+        }
+    }
+
+    /**
+     * Gets the current index by tab item ID
+     */
+    private getCurrentIndexById = (item: JSX.Element): boolean => {
+        return item.props.id === this.state.activeId;
+    }
+
+    /**
+     * Gets the child by the slot property
+     */
+    private getChildBySlot(children: React.ReactNode, slot: TabSlot): JSX.Element[] {
```

Comment:
I feel like `getChildBySlot` could be a generic utility - slots are something we use somewhat frequently

Reply:
Very true, I think we should have an issue for it though